src/task.c

```diff
@@ -471,124 +517,73 @@ task_worker_thread(void *arg)
+   platform_condvar_broadcast(&group->cv);
+   platform_condvar_unlock(&group->cv);
 
    // Allow all background threads to wrap up their work.
    for (uint8 i = 0; i < num_threads; i++) {
       platform_thread_join(group->bg.threads[i]);
+      group->bg.num_threads--;
    }
 }
 
 static void
 task_group_deinit(task_group *group)
 {
    task_group_stop_and_wait_for_threads(group);
-   if (task_system_use_bg_threads(group->ts)) {
-      platform_condvar_destroy(&group->bg.cv);
-   } else {
-      platform_mutex_unlock(&group->fg.mutex);
-      platform_mutex_destroy(&group->fg.mutex);
-   }
+   platform_condvar_destroy(&group->cv);
 }
 
 static platform_status
 task_group_init(task_group  *group,
                 task_system *ts,
                 bool         use_stats,
-                bool         use_bg_threads,
                 uint8        num_bg_threads,
                 uint64       scratch_size)
 {
    ZERO_CONTENTS(group);
    group->ts            = ts;
    group->use_stats     = use_stats;
-   platform_heap_id hid = group->ts->heap_id;
+   platform_heap_id hid = ts->heap_id;
    platform_status  rc;
-   if (use_bg_threads) {
-      group->bg.num_threads = num_bg_threads;
 
-      rc = platform_condvar_init(&group->bg.cv, hid);
-      if (!SUCCESS(rc)) {
-         return rc;
-      }
+   rc = platform_condvar_init(&group->cv, hid);
+   if (!SUCCESS(rc)) {
+      return rc;
+   }
 
-      for (uint8 i = 0; i < num_bg_threads; i++) {
-         rc = task_thread_create("splinter-bg-thread",
-                                 task_worker_thread,
-                                 (void *)group,
-                                 scratch_size,
-                                 ts,
-                                 hid,
-                                 &group->bg.threads[i]);
-         if (!SUCCESS(rc)) {
-            task_group_stop_and_wait_for_threads(group);
-            goto out;
-         }
-      }
-   } else {
-      rc = platform_mutex_init(&group->fg.mutex, 0, hid);
+   for (uint8 i = 0; i < num_bg_threads; i++) {
+      rc = task_thread_create("splinter-bg-thread",
+                              task_worker_thread,
+                              (void *)group,
+                              scratch_size,
+                              ts,
+                              hid,
+                              &group->bg.threads[i]);
       if (!SUCCESS(rc)) {
+         task_group_stop_and_wait_for_threads(group);
          goto out;
       }
+      group->bg.num_threads++;
    }
    return STATUS_OK;
 
 out:
    debug_assert(!SUCCESS(rc));
-   if (use_bg_threads) {
-      platform_condvar_destroy(&group->bg.cv);
-   }
+   platform_condvar_destroy(&group->cv);
    return rc;
 }
 
-/*
- * task_lock_task_queue(), task_unlock_task_queue():
- *
- * Primitives to lock / unlock task queues depending on whether we
- * are dealing with foreground or background threads.
- */
-static inline platform_status
-task_lock_task_queue(task_group *group)
```

Comment:
I understand that you rework makes these two lock / unlock functions unnecessary. 

However, would you consider retaining these shell inline functions as wrappers to `platform_mutex_lock/unlock()` interfaces, so that one can set a debugger breakpoint in these wrappers?